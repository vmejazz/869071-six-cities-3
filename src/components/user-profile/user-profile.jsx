import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {getUser} from "../../reducer/user/selectors.js";

const UserProfile = (props) => {
  const {userInfo} = props;
  let authStatus = userInfo.authorizationStatus === `AUTH` ? true : false;

  return (
    <React.Fragment>
      <div className="header__nav-link header__nav-link--profile">
        <div className="header__avatar-wrapper user__avatar-wrapper">
        </div>
        <span className=
          {`${authStatus ? `header__user-name user__name` : `header__login`}`}
        >
          <Link to={`${authStatus ? `/favorites` : `/login`}`}>
            { authStatus ? userInfo.email : `Sing in`}
          </Link>
        </span>
      </div>
    </React.Fragment>
  );
};

UserProfile.propTypes = {
  userInfo: PropTypes.shape({
    authorizationStatus: PropTypes.string,
    bookmarksRequired: PropTypes.bool,
    setCheckedStatus: PropTypes.bool,
    showRequestModal: PropTypes.bool,
    isCheckedStatus: PropTypes.bool,
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatarUrl: PropTypes.string,
    isPro: PropTypes.bool
  })
};

const mapStateToProps = (state) => ({
  userInfo: getUser(state)
});

export {UserProfile};
export default connect(mapStateToProps)(UserProfile);

