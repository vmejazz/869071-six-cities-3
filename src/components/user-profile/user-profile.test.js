import React from "react";
import renderer from "react-test-renderer";
import {UserProfile} from "./user-profile.jsx";
import customHistory from "../../history.js";
import {Router} from "react-router-dom";

it(`UserProfile section render`, () => {
  const tree = renderer
    .create(
        <Router history={customHistory}>
          <UserProfile
            userInfo={{authorizationStatus: `AUTH`}}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
