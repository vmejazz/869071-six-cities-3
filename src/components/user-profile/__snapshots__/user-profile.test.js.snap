// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile section render 1`] = `
<div
  className="header__nav-link header__nav-link--profile"
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />
  <span
    className="header__user-name user__name"
  >
    <a
      href="/favorites"
      onClick={[Function]}
    />
  </span>
</div>
`;
